rapids-cmake/test/generate_resource_spec.cmake

```diff
@@ -67,10 +67,12 @@ function(rapids_test_generate_resource_spec DESTINATION filepath)
   set(error_file ${PROJECT_BINARY_DIR}/rapids-cmake/detect_gpus.stderr.log)
 
   if(NOT EXISTS "${eval_exe}")
-    find_package(CUDAToolkit REQUIRED)
+    find_package(CUDAToolkit)
     file(MAKE_DIRECTORY "${PROJECT_BINARY_DIR}/rapids-cmake/")
 
-    set(compile_options "-I${CUDAToolkit_INCLUDE_DIRS}")
+    if(CUDAToolkit_FOUND)
+      set(compile_options "-I${CUDAToolkit_INCLUDE_DIRS}" "-DHAVE_CUDA")
+    endif()
```

Comment:
I get many warnings building with Apple `clang` (it stupidly still defaults to C++98):
```
_deps/rapids-cmake-src/rapids-cmake/test/detail/generate_resource_spec.cpp:69:11: error: no matching constructor for initialization of 'version'
  version v{1, 0};
          ^~~~~~~
_deps/rapids-cmake-src/rapids-cmake/test/detail/generate_resource_spec.cpp:27:13: warning: default member initializer for non-static data member is a C++11 extension [-Wc++11-extensions]
  int minor = 0;
            ^
_deps/rapids-cmake-src/rapids-cmake/test/detail/generate_resource_spec.cpp:31:22: error: expected member name or ';' after declaration specifiers
  gpu(int i) : id{i} {};
                     ^
_deps/rapids-cmake-src/rapids-cmake/test/detail/generate_resource_spec.cpp:32:33: error: expected member name or ';' after declaration specifiers
  gpu(int i, size_t mem) : id{i}, memory{mem}, slots{100} {}
                                ^
_deps/rapids-cmake-src/rapids-cmake/test/detail/generate_resource_spec.cpp:34:17: warning: default member initializer for non-static data member is a C++11 extension [-Wc++11-extensions]
  size_t memory = 0;
                ^
_deps/rapids-cmake-src/rapids-cmake/test/detail/generate_resource_spec.cpp:35:17: warning: default member initializer for non-static data member is a C++11 extension [-Wc++11-extensions]
  int slots     = 0;
                ^
_deps/rapids-cmake-src/rapids-cmake/test/detail/generate_resource_spec.cpp:31:18: error: expected '('
  gpu(int i) : id{i} {};
                 ^
```
This is fixed by the following
```suggestion
    else()
      set(compile_options "-std=c++14")
    endif()
```
Perhaps better to add this flag also for the GPU build.

Reply:
Corrected these errors by making sure the source is c++98 conformant. That way we can continue to support compilers that don't handle `-std=` flags